Revit_Core_Engine/Create/View/ViewPlan.cs

```diff
@@ -79,24 +88,135 @@ public static View ViewPlan(this Document document, Autodesk.Revit.DB.Level leve
             
             if (!string.IsNullOrEmpty(viewName))
             {
+                int number = 0;
+                string uniqueName = viewName;
+
+                while (uniqueName.IsExistingViewName(document))
+                {
+                    number++;
+                    uniqueName = $"{viewName} ({number})";
+                }
+
+#if (REVIT2018 || REVIT2019)
+                    newView.ViewName = uniqueName;
+#else
+                newView.Name = uniqueName;
+#endif
+                if (uniqueName != viewName)
+                {
+                    BH.Engine.Base.Compute.RecordWarning($"There is already a view named '{viewName}'. It has been named '{uniqueName}' instead.");
+                }
+            }
+            
+            return newView;
+        }
+
+        /***************************************************/
+
+        [Description("Creates and returns a new Floor Plan view in the current Revit file.")]
+        [Input("document", "The current Revit document to be processed.")]
+        [Input("level", "The level that the created Floor Plan refers to.")]
+        [Input("viewName", "Name of the new view.")]
+        [Input("viewFamily", "View Family of the View Type. The default is FloorPlan.")]
+        [Input("scopeBox", "(Optional) A Scope Box to attempt to apply to the newly created view.")]
+        [Input("viewTemplateId", "(Optional) A View Template Id to be applied in the view.")]
+        [Input("cropRegionVisible", "True if the crop region should be visible.")]
+        [Input("annotationCrop", "True if the annotation crop should be visible.")]
+        [Output("newView", "The new view.")]
+        public static View ViewPlan(this Document document, Level level, string viewName, ViewFamily viewFamily = ViewFamily.FloorPlan,  ElementId scopeBoxId = null, ElementId viewTemplateId = null, bool cropRegionVisible = false, bool annotationCrop = false)
```

Comment:
Now when I look at this method and the other one that creates view plan - shouldn't both have same inputs, the only difference being scope box vs crop box? So both would have `viewDetailLevel`, `viewFamily`, `cropRegionVisible`, `annotationCrop` etc. Then it would become one single method more or less, only outline being a different query for each. Or can't it be simplified to this extent?

Reply:
I will let myself play the devil's advocate one more time 😈 

1. You would have 2 methods, one taking `CurveLoop` and the other `Element`, all other inputs being the same - but then there would be a query extracting the outline of loop and scope box, to be passed to a common method taking the extracted outline and other inputs. Would that work?
2. You could always leave it empty for your purpose, am I correct?
3. I think there is not too many calls to it, should be easy enough to update the code and add versioning

Reply:
I still think it is not good idea to merge it like this because `Scope Box` and `Crop Box` are totally different:
- to set `Scope Box` you need to change `BuiltInParameter.VIEWER_VOLUME_OF_INTEREST_CROP` parameter
- to set `Crop Box` you need to use `SetCropShape()` from `ViewCropRegionShapeManager`
- If you use `Scope Box`, you can't change `Crop Box`

I would stay with two separate methods that will be used in different cases.

Reply:
How about having the new method call the existing method with `cropBox = null`, then apply the scope box and anything extra? 

To make that work, we'll probably need to add the `ViewFamily viewFamily = ViewFamily.FloorPlan` input to the existing method though, and do versioning... 😅 

![image](https://user-images.githubusercontent.com/102604891/222744272-b4fa5601-b8c4-43eb-9b3a-8f4125f09aa2.png)

Reply:
Unless you only do the `if (!(viewFamily == ViewFamily.FloorPlan || viewFamily == ViewFamily.CeilingPlan || viewFamily == ViewFamily.AreaPlan || viewFamily == ViewFamily.StructuralPlan))` check in the new method. In that case no need to do versioning?

Reply:
I combined these methods and added `viewFamily` input to existing. In the new method I call the existing one with `cropBox = null`.